state_test.go

```diff
@@ -180,7 +180,191 @@ func TestGatherNextStateInputs(t *testing.T) {
 	mdel := NewMockApplicationIntegration(t)
 
 	ap := New(mraft, mdel, withTimeProvider(mtime))
-	ap.startTime = time.Date(2020, 11, 2, 12, 0, 0, 0, time.UTC)
+	firstStateTime := time.Date(2020, 11, 2, 12, 0, 0, 0, time.UTC)
```

Comment:
Should these three tests be a table test? It took a minute to figure out what was different between these three

Reply:
Sure. Copy paste was quicker but I have tableified them now.